docs/compilers/CSharp/Nullability Public API Design Notes.md

```diff
@@ -84,75 +93,82 @@ It will share an initial-binding `BoundNode` cache with the standard semantic mo
 We add the new APIs for retreiving and interacting with nullability:
 
 ```C#
-public enum Nullability
+internal enum NullableAnnotation : byte
 {
-    NotApplicable,
-    NotComputed,
+    Default,
     Unknown,
+    NotAnnotated,
+    Annotated
+}
+
+internal enum NullableFlowState : byte
+{
+    Default,
     NotNull,
-    MayBeNull
+    MaybeNull
 }
```

Comment:
So there is no state for certainly null?

Reply:
Correct. We don't treat maybe-null and certainly-null any differently, so we only have one category ("this value is dangerous").

---
In reply to: [262742919](https://github.com/dotnet/roslyn/pull/33874#discussion_r262742919) [](ancestors = 262742919)

Reply:
Interesting. It seems for code fixers this would have been valuable; in the same way that exposing the folded constant is useful (e.g. replace with `null`).

Reply:
It might indeed be useful, but we don't calculate this in the compiler. Really what you want there is an actual points-to analysis, because you wouldn't just want to know whether it's certainly null but you'd also want to know what else it could be. @mavasani and @genlu have been working on that as part of making our IOperation-based dataflow analysis framework consumable by analyzer authors, so that's the route you'd want to take if you want that kind of information.